timemachine/ff/handlers/utils.py

```diff
@@ -67,3 +91,99 @@ def match_smirks(mol, smirks):
         matches.append(tuple(mas))
 
     return matches
+
+
+def check_bond_smarts_symmetric(bond_smarts: str) -> bool:
+    """Match [<atom1>:1]*[<atom2>:2]
+    and return whether atom1 and atom2 are identical strings
+
+    Notes
+    -----
+    * The AM1CCC model contains symmetric patterns that must be assigned 0 parameters
+        (Otherwise, undefined behavior when symmetric bond matches in an arbitrary direction)
+    * Only checks string equivalence!
+        for example
+        check_bond_smarts_symmetric("[#6,#7:1]~[#7,#6:2]")
+        will be a false negative
+    * Does not handle all possible bond smarts
+        for example
+        "[#6:1]~[#6:2]~[#1]"
```

Comment:
Already a problem: there are two patterns in the current model that are not parsed correctly

`[#6X3:1](~[#8X1,#16X1])(~[#16X1:2])`
and
`[#6X3:1](~[#8X1,#16X1])(~[#8X1:2])`

Since these don't match the regex, they return `False` by default.

Don't have a sense of what SMARTS subset may be exercised by future edits to this force field.

Reply:
What about using the following pattern?

```
pattern = re.compile(r"(\[(?P<atom1>.+)\:1\].+\[(?P<atom2>.+)\:2\]`)
...
complete = match.span() == (0, bond_smarts.index(":2]") + 3)
```

Would cover those two cases. Unless you also need to match the `16X1` to the `16X1` in the middle? Which I guess you could do the hacky thing of having a third group in the middle and check if the atom1 and atom2 are represented somewhere in the middle....